Artifactory trigger, Setup / External Configuration updates (#1653)
spinnaker/spinnaker.github.io: setup/configuration/index.md

This page describes Spinnaker's external configuration feature. Using external configuration, you can manage account configuration (either complete account configuration, or only secrets such as account passwords) externally from Spinnaker.

> External configuration uses the open-source [Spring Cloud Config](https://spring.io/projects/spring-cloud-config) project, which includes a configuration server and client libraries to support centralized external configuration in a distributed system. The Config Server connects to a remote configuration repository and serves stored configuration properties to client applications. Configuration properties can be organized by Spring profile and application name and can be stored in any of a number of backends, including Git, HashiCorp Vault, AWS S3, JDBC, CredHub, and others. For more information about Spring Cloud Config, see the [documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html).

Reviewer: ```suggestion
External configuration uses the open-source [Spring Cloud Config](https://spring.io/projects/spring-cloud-config) project, which includes a configuration server and client libraries to support centralized external configuration in a distributed system. The Config Server connects to a remote configuration repository and serves stored configuration properties to client applications. 

Configuration properties can be organized by Spring profile and application name and can be stored in any of a number of backends, including GitHub, HashiCorp Vault, AWS S3, JDBC, CredHub, and others. For more information about Spring Cloud Config, see the [documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html).
```

          token: [vault access token]
```

> The above example configures authentication for Vault using the Vault
> `X-Config-Token` header, via the setting of the

> > `spring.cloud.config.server.vault.token` property. In Spinnaker 1.18, you  can

Reviewer: ```suggestion
> `spring.cloud.config.server.vault.token` property. Starting with Spinnaker 1.18, you  can
```

Reviewer 2: replace
`using the Vault X-Config-Token header, via the setting ...`
with
`using a static authentication token, via the setting ...`

> use other methods of configuring authentication for Vault. See the relevant
> [Spring Cloud Config Server
> documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/reference/html/#vault-backend)
> for more information.

### Using an S3 backend

To use an AWS S3 backend, enable the `awss3` profile and configure the settings under `spring.cloud.config.server.awss3`. Your configuration might look like the following example:

```yml
spring:
  profiles:
    active: awss3
  cloud:
    config:
      server:
        awss3:
          region: us-west-2
          bucket: my-bucket
```

To configure authentication, you must use AWS's Default Credential Provider Chain. See the [AWS SDK documentation](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html) for more information.

> For information about configuring other supported Config Server backends, see the [Spring Cloud Config Server documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html#_environment_repository).

Reviewer: ```suggestion
For information about configuring other supported config-server backends, see the [Spring Cloud Config Server documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html#_environment_repository).
```

Author: I think I use "Config Server" generally as a shortened form of the project name (Spring Cloud Config Server), less as a generic term for something that serves configuration. Is that use OK?

> The Kubernetes, Google Cloud, and App Engine cloud providers can load account information from files that are separate from the service YAML configuration. You can load these separate files from an external source using the Config Server's Resource abstraction. In the service YAML configuration, prefix a file path with `configserver:` to indicate that the file should be retrieved by the Config Server. 

Reviewer: There's no Google Cloud provider, per se. There's Compute Engine, App Engine (which is listed here), and Kubernetes Engine. I assume this is a reference to Compute Engine?

Author: Fixed. Thanks!
